lib/script/common.js

```diff
@@ -414,6 +414,12 @@ exports.hashType = {
 
   SINGLE: 3,
 
+  /*
+   * Sign output at the n-i'th index (zero sequences).
+   */
+
+  SINGLEREVERSE: 6,
```

Comment:
I chose 6 to pad some space to allow porting of sighash types that come from Bitcoin more easily. Even though handshake is quite different from Bitcoin in some ways, its easier when things are more uniform. This may not be necessary and we could just use `4` because it impacts the `isSignatureEncoding` function below. Sighash types `4` and `5` will not cause this to return `false`:
https://github.com/handshake-org/hsd/pull/198/files#diff-4764bef9dd683c153cdffd9d3c06cf29R510

```javascript
if (type < exports.hashType.ALL || type > exports.hashType.SINGLEREVERSE)
```

Reply:
I think we should either disallow 4 and 5 or make SINGLEREVERSE 3. Allowing 4 and 5 introduces malleability. I'm actually partial to the latter, but willing to go with either.

Thinking out loud here: when I was first looking at this, I thought for a second that we should make 4 or 5 disallowed by policy rules, but now I don't think it matters. This is a consensus check on handshake (as opposed to bitcoin where it's policy), but new sighash types can't actually be soft-forked in without a new witness version (or new opcode), so we don't really gain anything by making it policy only. We can disallow 4 and 5 by consensus rules and still add them in later with a new witness version.

Reply:
Hmm, on second thought this actually doesn't introduce malleability since the sighash itself commits to the hash type. I suppose bitcoin disallows it just because they want everything "weird" off the policy layer.

Reply:
> or make SINGLEREVERSE 3

Do you mean 4 here? `SINGLE` is already 3.

After thinking about it some more, skipping some numbers for the sighash types gives more headache than the small benefit gained from being able to more easily port Bitcoin based software to Handshake.  Who knows if/when Bitcoin will add new sighash types, its a definite premature optimization.

Just pushed up new commits that make `SINGLEREVERSE` 4

Reply:
Yeah, sorry, 4 is what I meant. Not thinking clearly.